modules/auxiliary/scanner/misc/cisco_smart_install.rb

```diff
@@ -57,10 +132,23 @@ def smi?
     end
   end
 
-  def run_host(_ip)
+  def run_host(ip)
+    @lhost    = datastore['LHOST'] || Rex::Socket.source_address(ip)
     begin
-      connect
-      return unless smi?
+      case
+        when action.name == 'SCAN'
+          connect
+          return unless smi?
+        when action.name == 'DOWNLOAD'
+          start_tftp
+          connect
+          packet = craft_packet
```

Comment:
Any reason to not `smi?` beforehand?

Reply:
I originally did not cause sometimes it forces it into a temporarry DOS state, but i guess I could just have it sleep temporarily.